pkg/dns/service.go

```diff
@@ -323,3 +324,26 @@ func (s *Service) CleanupDNSRecords(ctx context.Context, owner traffic.Interface
 	}
 	return nil
 }
+
+// getDNSRecordManagedZone returns the current ManagedZone for the given DNSRecord.
+func (s *Service) GetDNSRecordManagedZone(ctx context.Context, dnsRecord *v1alpha1.DNSRecord) (*v1alpha1.ManagedZone, error) {
+
+	if dnsRecord.Spec.ManagedZoneRef == nil {
+		return nil, fmt.Errorf("no managed zone configured for : %s", dnsRecord.Name)
+	}
+
+	managedZone := &v1alpha1.ManagedZone{}
+
+	err := s.controlClient.Get(ctx, client.ObjectKey{Namespace: dnsRecord.Namespace, Name: dnsRecord.Spec.ManagedZoneRef.Name}, managedZone)
+	if err != nil {
+		return nil, err
+	}
+
+	managedZoneReady := meta.IsStatusConditionTrue(managedZone.Status.Conditions, "Ready")
```

Comment:
wondering why we check this here? I feel like whatever is using the Zone should do this check

Reply:
thats a good question im not sure. that change was merged a few days ago as part of the bug Mike found.

Reply:
This shouldn't be here, where this method was originally should still be doing it though https://github.com/Kuadrant/multicluster-gateway-controller/pull/283/files#diff-6321f40c52e93a933e0ad7d590b390c749c8e6b757a10b40bf4d21b9a4fa4603R167

Reply:
i moved the original function but ill move the is status condition back